src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java

```diff
@@ -214,4 +214,50 @@ public void setConnectionId(String connectionId) {
 
   @JsonRequest("sonarlint/getRemoteProjectNames")
   CompletableFuture<Map<String, String>> getRemoteProjectNames(GetRemoteProjectsNamesParams params);
+
+  class GetServerPathForTokenGenerationResponse {
+    private String serverUrl;
+    private  String errorMessage;
+
+    public GetServerPathForTokenGenerationResponse(String serverUrl, String errorMessage) {
+      setServerUrl(serverUrl);
+      setErrorMessage(errorMessage);
+    }
+
+    public String getServerUrl() {
+      return serverUrl;
+    }
+
+    public void setServerUrl(String serverUrl) {
+      this.serverUrl = serverUrl;
+    }
+
+    public String getErrorMessage() {
+      return errorMessage;
+    }
+
+    public void setErrorMessage(String errorMessage) {
+      this.errorMessage = errorMessage;
+    }
+
+  }
+
+  class GetServerPathForTokenGenerationParams {
```

Comment:
Is this class needed? I believe it should be possible to pass a `String` directly (and avoid the boilerplate 😛)

Reply:
As discussed, this is the pattern we use for this parameters and it's for the consistency of the code style. We can decide later if we want to refactor it.